src/tests/test_error_messages.py

```diff
@@ -0,0 +1,179 @@
+import pytest
+
+from sqlalchemy.exc import OperationalError
+from IPython.core.error import UsageError
+
+
+@pytest.fixture
+def query_no_suggestion():
+    return (
+        "sql",
+        "",
+        """
+        sqlite://
+        SELECT FROM author;
+        """,
+    )
+
+
+@pytest.fixture
+def query_incorrect_column_name():
+    return (
+        "sql",
+        "",
+        """
+        sqlite://
+        SELECT first_(name FROM author;
+        """,
+    )
+
+
+@pytest.fixture
+def query_suggestion():
+    return (
+        "sql",
+        "",
+        """
+        sqlite://
+        ALTER TABLE author RENAME new_author;
+        """,
+    )
+
+
+@pytest.fixture
+def query_multiple():
+    return (
+        "sql",
+        "",
+        """
+        sqlite://
+        INSERT INTO author VALUES ('Charles', 'Dickens', 1812);
+        ALTER TABLE author RENAME another_name;
+        """,
+    )
+
+
+def test_syntax_error_no_suggestion(ip, capsys, query_no_suggestion):
+    with pytest.raises(UsageError) as err:
+        ip.run_cell_magic(
+            *query_no_suggestion,
+        )
+
+    out, _ = capsys.readouterr()
+    assert '(sqlite3.OperationalError) near "FROM": syntax error' in out
+    assert "Looks like there is a syntax error in your query" in str(err)
+    assert (
+        "If you need help solving this issue, "
+        "send us a message: https://ploomber.io/community" in str(err)
+    )
+
+
+def test_syntax_error_incorrect_column_name(ip, capsys, query_incorrect_column_name):
+    with pytest.raises(UsageError) as err:
+        ip.run_cell_magic(
+            *query_incorrect_column_name,
+        )
+    out, _ = capsys.readouterr()
+    assert '(sqlite3.OperationalError) near "FROM": syntax error' in out
+    assert "Looks like there is a syntax error in your query" in str(err)
+    assert (
+        "Syntax Error in SELECT first_(name FROM author: "
+        "Expecting ) at Line 1, Column 24" in str(err)
+    )
+    assert (
+        "If you need help solving this issue, "
+        "send us a message: https://ploomber.io/community" in str(err)
+    )
+
+
+def test_syntax_error_suggestion(ip, capsys, query_suggestion):
+    with pytest.raises(UsageError) as err:
+        ip.run_cell_magic(
+            *query_suggestion,
+        )
+    out, _ = capsys.readouterr()
+    assert "Did you mean : ['ALTER TABLE author RENAME TO new_author']" in str(err)
+    assert "Looks like there is a syntax error in your query" in str(err)
+    assert (
+        "If you need help solving this issue, "
+        "send us a message: https://ploomber.io/community" in str(err)
+    )
```

Comment:
I think we can put these 2 messages in a global variable since they might change in the future and fail our tests.

Reply:
Done. fetching from ploomber_core now